### ISSUE TITLE
Bad monitoring metrics with integration callbacks are difficult to notice.

### ISSUE THREAD

**Issue by hvy** [optuna/optuna#1226]
## Motivation

Bad/wrong monitoring metrics represented by key strings with the following integration callbacks are difficult to notice. For instance, a simple typo in the metric would make the pruner a NOOP pruner. One might expect an error.

- https://github.com/optuna/optuna/blob/master/optuna/integration/keras.py (resolved by #1759)
- https://github.com/optuna/optuna/blob/master/optuna/integration/tfkeras.py (resolved by #2156)
- https://github.com/optuna/optuna/blob/master/optuna/integration/pytorch_lightning.py (resolved by #2157, ongoing)

## Description

Above callbacks are not doing dictionary lookups that raise on failure but uses `dict.get`. I'm wondering if we cannot raise instead as that'd clearly communicate to the user that the pruner isn't doing anything. However, I'm not too familiar with all of the above libraries and it might not be possible.

**Comment by hvy** [comment]
This is still an issue.

**Comment by toshihikoyanase** [comment]
Keras pruning callback was fixed by @bigbird555 in #1759.
The callbacks of [tfkeras](https://github.com/optuna/optuna/blob/master/optuna/integration/tfkeras.py) and [pytorch lightning](https://github.com/optuna/optuna/blob/master/optuna/integration/pytorch_lightning.py) are still waiting for your contribution.

**Comment by nzw0301** [comment]
@bigbird555 Kudos to your all contributions to this issue! 

---


Do we have something to do related to this issue? Otherwise, I suppose that we can close this isseu.

**Comment by hvy** [comment]
You're right, let me close it. Thanks @nzw0301 for the heads up and again, thanks @bigbird555 !
